src/expr/src/lib.rs

```diff
@@ -13,6 +13,7 @@
 // limitations under the License.
 
 #![allow(rustdoc::private_intra_doc_links)]
+#![allow(non_snake_case)] // for `ctor` generated code
```

Comment:
Haven't found a better way to only apply it to `ctor`. The code below still does not work. 😕

```rust
#[allow(non_snake_case)]
const _: () = {
    #[ctor]
    fn foo() { }
};
```